jsonnet/kube-thanos/kube-thanos-servicemonitors.libsonnet

```diff
@@ -0,0 +1,58 @@
+{
+  thanos+:: {
+    querier+: {
+      serviceMonitor+: {
+        apiVersion: 'monitoring.coreos.com/v1',
+        kind: 'ServiceMonitor',
+        metadata+: {
+          name: 'thanos-querier',
+          namespace: 'monitoring',
+        },
+        spec: {
+          selector: {
+            matchLabels: $.thanos.querier.service.metadata.labels,
+          },
+          endpoints: [
+            { port: 'http' },
+          ],
+        },
+      },
+    },
+    store+: {
+      serviceMonitor+: {
+        apiVersion: 'monitoring.coreos.com/v1',
+        kind: 'ServiceMonitor',
+        metadata+: {
+          name: 'thanos-store',
+          namespace: 'monitoring',
+        },
+        spec: {
+          selector: {
+            matchLabels: $.thanos.store.service.metadata.labels,
+          },
+          endpoints: [
+            { port: 'http' },
+          ],
+        },
+      },
+    },
+    receive+: {
+      serviceMonitor+: {
+        apiVersion: 'monitoring.coreos.com/v1',
+        kind: 'ServiceMonitor',
+        metadata+: {
+          name: 'thanos-receive',
+          namespace: 'monitoring',
```

Comment:
is hardcoding the namespace here intended?

Reply:
Yes. All objects have hardcoded namespaces. We don't use `_config+::{}` in this project at all (except in the thanos-mixin that I started in the other PR).
One needs to patch those namespaces in their downstream `main.jsonnet`.